bc_events/session.py

```diff
@@ -110,6 +110,32 @@ def publish(self, action=None, entity=None, data=None, category=None):
         topic = self.client.get_topic(category, entity, action)
         self._publish(topic, data)
 
+    def publish_all(self, events):
+        """Publish all events
+
+        Parameters
+        ----------
+        events : list
+            A list of events to publish
+        """
+
+        # If there are less than BULK_EVENT_SINGLE_POST_THRESHOLD events in the queue, publish individually
+        if len(self.events) <= BULK_EVENT_SINGLE_POST_THRESHOLD:
+            for event in self.events:
+                event.publish()
+            return
+
+        if not self.client.publish_url:
+            return
+
+        all_event_data = [event.request_json for event in events]
+
+        # Publish up to MAX_BULK_EVENTS events at a time
+        for i in range(0, len(all_event_data), MAX_BULK_EVENTS):
+            event_data = all_event_data[i : i + MAX_BULK_EVENTS]
+            logger.info("Publishing {0} Events".format(len(event_data)), extra={"context": event_data})
+            requests.post(self.client.publish_all_url, json=event_data)
```

Comment:
I see value in being transparent in the logs about the chunking of events into separate requests. However, having the logging statement here means that tests and local development won't see these logs (since we `return` a few lines up).

Perhaps, instead of returning early above, we do:
```suggestion
            if self.client.publish_all_url:
                requests.post(self.client.publish_all_url, json=event_data)
```